lib/webapp.js

```diff
@@ -134,6 +149,51 @@ DataserviceContext.prototype = {
   makeErrorObject: zluxUtil.makeErrorObject
 };
 
+function createDataserviceStorage() {
+  return process.clusterManager.getStorage().then( function(clusterStorage) {
+    
+    clusterStorage.getStorage = function () {
+      return process.clusterManager.getStorage().then((storage) => {
+        clusterStorage.storageDict = storage.storageDict;
+        return storage.storageDict;
+      });
+    }
+
+    clusterStorage.setStorageValue = function (id, value) {
+      clusterStorage.storageDict[id] = value;
+      process.clusterManager.setStorageValue(id, value);
+    }
+
+    clusterStorage.setStorage = function (dict) {
+      clusterStorage.storageDict = dict;
+      process.clusterManager.setStorage(dict);
+    }
+
+    clusterStorage.addStorage = function (dict) {
+      clusterStorage.storageDict = { ...clusterStorage.storageDict, ...dict };
```

Comment:
Have you tested how this merges objects with depth n > 2,
aka would:

```
let obj1 = {
     key1: 'value1',
     key2: {
       subkey1: 'value2'
     }
}
```

merge properly with

```
let obj2 = {
     key1: 'differentValue',
     key2: {
          subkey2: {
                subkey3: 'value3'
           }
     }
}
```

Reply:
@timgerstel That's a good point, and if I were to guess I would say it wouldn't handle this because I remember @1000TurquoisePogs having a problem doing this and not being able to find a JS way (and had to import some package?)

Reply:
I figured it out. Will add my own method to do a recursive merge

Reply:
I'm just not sure why this function exists, though.